pandas/core/arrays/datetimelike.py

```diff
@@ -1823,6 +1823,37 @@ def validate_periods(periods):
     return periods
 
 
+def validate_inclusiveness(inclusive):
```

Comment:
maybe just call this validate_inclusive?

Reply:
couple of things. I would like to move this new routine & validate_entpoint to pandas/core/indexers/utils.py i think.
Also need to slightly refactor `.between_time` (which was just merged) to use the same.

Might be best to do these as a pre-cursor PR to this one.

Reply:
might also look at other merged prs from underlying issue that may be able to use this validator,